### PR TITLE
Extract token array shape into a Psalm type

### PR DESCRIPTION
**Pull request by greg0ire**
This should make things less error prone and easier on the eyes.

### CODE REVIEW

File: lib/Doctrine/Common/Lexer/AbstractLexer.php [hunk 1]
```diff
@@ -160,7 +162,7 @@ public function isNextToken($token)
     /**
      * Checks whether any of the given tokens matches the current lookahead.
      *
-     * @param string[] $tokens
+     * @param list<int|string|null> $tokens
```
Comment (greg0ire) [inline lib/Doctrine/Common/Lexer/AbstractLexer.php]: I don't know if it makes sense to allow `null` here, but I'm forced to do so if a token type can be passed here. Either this is correct or there should be a subtype of `Token` that disallows `null`. Maybe we should also rename `$tokens` to `$tokenTypes` to make things less confusing.

File: lib/Doctrine/Common/Lexer/AbstractLexer.php [hunk 2]
```diff
@@ -18,6 +18,8 @@
 
 /**
  * Base class for writing simple lexers, i.e. for creating small DSLs.
+ *
+ * @psalm-type Token = array{value: int|string, type:string|int|null, position:int}
```
Comment (greg0ire) [inline lib/Doctrine/Common/Lexer/AbstractLexer.php:22]: I'm unsure if we should really allow `int` as a value, or if `dataProvider` should be fixed to quote the `10`

File: lib/Doctrine/Common/Lexer/AbstractLexer.php (hunk 1 above)
Comment (greg0ire) [reply lib/Doctrine/Common/Lexer/AbstractLexer.php]: Looking a bit more into the code, I think the type will be null if it is not understood by the lexer (no condition in `getType` matches the value). I don't think it makes sense to check if that's the case, so let me create another type just for that.

Comment (greg0ire) [reply lib/Doctrine/Common/Lexer/AbstractLexer.php]: > Maybe we should also rename $tokens to $tokenTypes to make things less confusing.

`skipUntil` uses `$type`, so let's be consistent with that.

**Review by derrabus — APPROVED** [review]

**Pull request merged**